app/components/chat.tsx

```diff
@@ -379,12 +379,13 @@ function useScrollToBottom() {
   // for auto-scroll
   const scrollRef = useRef<HTMLDivElement>(null);
   const [autoScroll, setAutoScroll] = useState(true);
-
+  const config = useAppConfig();
+  let isAutoScrollEnabled: boolean = config.autoScrollMessage;
   function scrollDomToBottom() {
     const dom = scrollRef.current;
     if (dom) {
       requestAnimationFrame(() => {
```

Comment:
Just info 
this `requestAnimationFrame` stupid complex that made performance in browser bad when there is lots of messages

Reply:
I don't know this repo much, but I think they've added this recently to make the show text animation smoother. If this impacts performance so bad, I guess it should be either an option or omitted.

Reply:
I was looking around it yesterday trying to Improve it